src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -4641,4 +4641,10 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_ConstantStringTooLong" xml:space="preserve">
     <value>Length of String constant exceeds current memory limit.  Try splitting the string into multiple constants.</value>
   </data>
+  <data name="ERR_ExpressionTreeContainsLocalFunction" xml:space="preserve">
+    <value>An expression tree may not contain a local function or a reference to a local function</value>
```

Comment:
I was thinking of both that if statements might be allowed in the future, as well as a hint to the coder that local functions in general are bad (if the message just said references to local functions, it might be a bit head-scratching).

Then again, if we do allow statements, a lot more things would need an overhaul and I assume we'd update the message, so I can change the current message to just say references.